### DISCUSSION TITLE
Help offered, help required

### DISCUSSION THREAD

**Discussion by lucamonfredo** [EdNekebno/DGTCentaurMods#10] [General]
Hi!

As a proud owner of a DGT Centaur board, I would gladly offer some help and time to move this project forward, but in turn I need some help setting thing up on my side 😉 .

My main goal would be to use a Raspberry Pi Zero W to be able to connect to the board via wifi and download (or implement some kind of exporting function) the PGN of the games, but I would also love to be able to use the board for online play.

Right now I tried to setup everything on a Pi Zero W with a Raspberry Pi OS Lite image, but I can't get the display to work, (everything else seems OK), can you help me solve this problem or put me in the right direction in order to actively working on features?

Is there some sort of roadmap? Can we maybe define it together?

Thanks for the great work on the reverse engineering of the board functions, looking forward to tinker with it!

**Comment by wormstein** [comment]
> Right now I tried to setup everything on a Pi Zero W with a Raspberry Pi OS Lite image, but I can't get the display to work, (everything else seems OK), can you help me solve this problem or put me in the right direction in order to actively working on features?

Hi and wekcome.
Have you enabled spi1.0 device? Add dtoverlay=spi1-3cs to config,txt file in boot partition,
Next, try to put the edp2in9d.py file in centaur folder and rename it to epaper.py so that centaur will find it in case you get file not found error.

I think there is another way to add edp driver but the solution above works for me,

Cjheers

**Comment by lucamonfredo** [comment]
Thanks @wormstein, 

I think my problem is more low level, or I am missing something really basic, because neither `centaur` binary nor `menu.py` seems to work (or, to be more clear, they seem to run without errors, but I just can't see anything on the display).

To get the scripts to work, I naturally had to install the necessary Python libraries (pyserial and Pillow) along with some system libraries (e.g. libtiff5), but I also had to configure my Pi to enable serial communication and disable bluetooth (via `/boot/config.txt`).

Having done this, I noticed that there are still some differences between the configuration of the Pi provided by DGT and mine based on Raspberry Pi OS Lite.

For example, in order for the board to respond to commands, I had to change the serial port from `/dev/ttyS0` to `/dev/ttyAMA0`, and now I'm stuck with the display that doesn't seem to respond.

Any ideas?
Thanks in advance!

**Reply by wormstein** [reply]
I never had to change anything for serial to work... just to enable it. It runs on /dev/ttyS0 just fine.

Is the centaur software working?

**Reply by lucamonfredo** [reply]
Now I'm running on `/dev/ttyAMA0`, I had to change the reference in `boardfunctions.py` to make everything work. I will try again starting from scratch and annotating everything that I do because I could have made some other things that I don't remember, I tinkered quite a bit ;)

**Reply by lucamonfredo** [reply]
BTW, centaur software seems to work OK.

**Comment by wormstein** [comment]
Enable SPI using raspi-config then enable second SPI slave device (/dev/spi1.0) in config.txt. See how in my previous comment.
Display uses SPI bus.

-Adrian

**Reply by lucamonfredo** [reply]
Great! I didn't notice you edited the reply, now I'm up and running, thanks a ton for the help! I'm preparing some additions to the README in order to have a basic guideline for others to join.
